spec/outputs/elasticsearch_spec.rb

```diff
@@ -306,6 +306,56 @@
     end
   end
 
+  describe "http client create" do
+    require "logstash/outputs/elasticsearch"
+    require "elasticsearch"
+
+    let(:options) {
+      {
+        "protocol" => "http",
+        "index" => "my-index",
+        "host" => "localhost",
+        "path" => "some-path"
+      }
+    }
+
+    let(:eso) {LogStash::Outputs::ElasticSearch.new(options)}
+
+    let(:manticore_host) {
```

Comment:
Instead of exposing the client through eso, maybe we just have a reader method that computes the URL based on the configuration?

Then we can do this:
```ruby
eso = LogStash::Outputs::ElasticSearch.new(...)
...
expect(eso.http_url).to(be == "http://whatever/something")
```

This feels closer to a unit test. If we want to test through Manticore, I feel like we should setup a mock http server to point Manticore at and make assertions about its behavior.

Thoughts?